### ISSUE TITLE
SD card seems not to be detected on ChromeBook

### ISSUE THREAD

**Issue by kelson42** [kiwix/kiwix-android#18]
It's possible to run Android app on ChromeBook. Unfortunately, it seems that the SD card is not detected. Here a feedback of a user (James)

> Under files I have "SD Card"
> I have now put "wikipedia_en_all_2016-12.zim" on the card.
> When I go to the SD card and click on that file Kiwix automatically opens but then says:
> "Error. The selected zim cannot be found".
> When I go to "storage" under settings within Kiwix all I get is "internal 20.2 GB".
> I am however able to set the chromebook so it automatically downloads new files via google chrome to the SD card.

**Comment by kelson42** [comment]
The "only" think we need to fix that problem is the full patch on the system on which the SD card is mount.

**Comment by mhutti1** [comment]
@kelson42 I don't think this will be that east... [chromium bug report](https://bugs.chromium.org/p/chromium/issues/detail?id=660189#c6). It looks like neither read nor write access to the SD card is freely available at the moment, and the later might not be for some time. We might be able to get a work around set up to get rid of the "Error. The selected zim cannot be found". but it likely is an issue with our native code not having the permissions to access the file URI. I doubt there is much we could do to resolve this without a lot of work.

**Comment by kelson42** [comment]
@mhutti1 Thx for the research and the input. That's sad, but we need to keep track on this problem. That said, I'll de-scope this ticket of 2.3.

**Comment by kelson42** [comment]
@julianharty Have you been able to read the SD (with kiwix-android) card on your chromebook?

**Comment by julianharty** [comment]
There _does_ seem to be a problem accessing the SD-Card with Kiwix-Android (Version 2.2 Build 53) on a Chromebook. I've tested the app on my ASUS Chromebook where I've got several ZIM files in the Downloads folder and in addition I've got a 64GB removable Micro SD card with a small ZIM file on it. The app cannot see anything on the SD-Card. It can see ZIM files in the Downloads folder. It cannot see them on the SD-Card. It also seems to have a private area where it downloads ZIM files (there are 4 I've downloaded currently).

I did try opening the ZIM file on the SD-Card (by double-clicking on the ZIM file)[1]. Kiwix is brought to the foreground but them pops up a short Android Toast message `Error: The selected ZIM file could not be found`. If and when I can find out how to access the Android Log on ChromeOS perhaps I'll find out more about how the path is presented. Also, perhaps we could add information to the Toast message to display the ZIM filename including the path and also more info on _why_ the ZIM file cannot be read by the app.

BTW: kiwix-js can access the files in Downloads and on the SD-card on Chrome OS.  

[1] Another quirk: If I double-click on a ZIM file in my Google Drive folders then neither Kiwix-Android nor kiwix-js are launched. Instead my Chromebook suggests I install various unrelated apps. It seems there's more to learn on how ChromeOS runs Android apps...

I'm happy to do more testing if/when there are some more things to do.
